exporter: swap pusher for exporter (#1656)
open-telemetry/opentelemetry-go: CHANGELOG.md

      "otel/sdk/trace".ReadOnlySpan
      "otel/sdk/trace".ReadWriteSpan
```

> - Swap "pusher" naming convention to "exporter" for consistent naming across codebase (#1656).

Reviewer: Can we update this to provide useful info to any user of the project?  Being explicit with the exported things that they will need to pay attention to. E.g.

```suggestion
- `"go.opentelemetry.io/sdk/metric/controller.basic"WithPusher` is replaced with `WithExporter` to provide consistent naming across project. (#1656)
```

Also, this needs to be moved to the `Unreleased` section.

Author: definitely, makes sense.
